### ISSUE TITLE
Image Reshaping with SVM doesn't work

### ISSUE THREAD

**Issue by mkosik** [Qualeams/Android-Face-Recognition-with-Deep-Learning-Test-Framework#27]
When trying to run a "Recognition Training" for Image Reshaping with SVM, it stucks on a white screen (Training Activity) & it doesn't process any image successfully.
The same happens for an app compiled from this repo with v1.5.3 lib (imported as a module, due to an arr corrupt problem) as well as currently published version on PlayStore.

@sladomic do you have any suggestion what might go wrong please?

Also Recognition View isn't handled properly for cases when it is attempted for reco method which isn't trained yet & it crashes the app.

Best regards,
Miro

**Comment by sladomic** [comment]
HI @mkosik 

> When trying to run a "Recognition Training" for Image Reshaping with SVM, it stucks on a white screen (Training Activity) & it doesn't process any image successfully.

With the current code, it's inefficient to use a high N for image reshaping. For the neural network of facenet we use 160 x 160 as input and get vector of 1 x 128 as output. But if you use the same number for image reshaping, you get 160 x 160 x 4 (channels) = 102'400

Here's the inefficient bottleneck in the SupportVectorMachine class

```java
public String getSvmString(Mat img){
        img = getFeatureVector(img);
        String result = "";
        for (int i=0; i<img.cols(); i++){
            result = result + " " + i + ":" + img.get(0,i)[0];
        }
        return result;
    }
```

**Comment by sladomic** [comment]
If you use N < 50, it works. We mostly used N = 25, because the results were similar to N = 50, but it was a lot faster.

**Comment by TomSang** [comment]
I have a small patch to fix this issue.  The performance of Reshaping with N=160 will be close to that of TensorFlow with SVM.

But I don't know  how to submit it.

**Comment by chithrabaskar** [comment]
by changing the package name of this app it did not work...why??

**Comment by sladomic** [comment]
@tomsang to submit your code you could 1. fork the repo, 2. implement the changes in the forked repo, 3. Create a pull request. I can then test and check it and then merge into the core repo.

**Comment by sladomic** [comment]
@chithrabaskar does Android Studio give you any errors? At which point does it crash?

**Comment by chithrabaskar** [comment]
No, there is no error by changing the package name...but it unfortunately stopped while running...please give me any solution

**Comment by TomSang** [comment]
I would like to provide a improve patch in the later
